napari/_qt/layer_controls/qt_image_controls_base.py

```diff
@@ -152,6 +152,9 @@ def __init__(self, layer: Image) -> None:
         self.colorbarLabel.setToolTip(trans._('Colorbar'))
 
         self._on_colormap_change()
+        if self.__class__ == QtBaseImageControls:
+            # this if is to prevent leaking of buttons in the tests
+            self.layout().addRow(self.button_grid)
```

Comment:
I don't understand this at all... Could you comment on what was happening before, and how this change fixes it?

Reply:
The `self.button_grid` is a layout that contains `self.panzoom_button` and `self.transform_button`, but the layout does not have a set parentship. So when try to detect a leaking widget, the `QApplication.topLevelWidgets()` is detecting this as a top-level widget without set parent. So it is handled by from `conftest.py`:

```
        if widget.parent() is not None:
            continue
```

This class and the second one are abstract class that we do not create instance outside the tests.

Reply:
> This class and the second one are abstract class that we do not create instance outside the tests.

ooooooh ok.

My next question is — why do we only add a parent in this circumstance rather than parenting to the layout always?

Reply:
No, because then in subclass we will need to undock first, that will create more messy code.